flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/optimize/program/FlinkDecorrelateProgram.scala

```diff
@@ -77,6 +81,15 @@ class FlinkDecorrelateProgram[OC <: FlinkOptimizeContext] extends FlinkOptimizeP
         join.getCondition.accept(visitor)
         super.visit(join)
       }
+
+      override def visit(other: RelNode): RelNode = {
+        other match {
+          // ignore Uncollect's inputs due to the correlate variables are from UNNEST directly,
+          // not from cases which RelDecorrelator handles
+          case r: Uncollect => r
```

Comment:
> not from cases which RelDecorrelator handles

I don't get it.
The real reason should be that `LogicalUnnestRule` is in the `DEFAULT_REWRITE` phase, and the `DEFAULT_REWRITE` phase is behind the `DECORRELATE` phase, so now we need not verify the Unnest correlation?
Update comments?

Reply:
`RelDecorrelator` only handles correlation variables in join, project and filter, does not handle UDTF pattern:`LogicalCorrelate` on `LogicalTableFunctionScan` which will be converted to `BatchExecCorrelate` or `StreamExecCorrelate`. while correlation variables in join, project and filter are not supported in physical phase.  so the validation in `FlinkDecorrelateProgram` only need to check whether there are correlation variables in join, project and filter.